cms/search/tests/test_publishers.py

```diff
@@ -0,0 +1,425 @@
+from cms.release_calendar.enums import ReleaseStatus
+from cms.release_calendar.tests.factories import ReleaseCalendarPageFactory
+from cms.search.publishers import BasePublisher, KafkaPublisher, LogPublisher
+from cms.standard_pages.models import InformationPage  # Uses GenericTaxonomyMixin
+from cms.standard_pages.tests.factories import IndexPageFactory, InformationPageFactory
+from cms.taxonomy.models import GenericPageToTaxonomyTopic, Topic
+
+EXPECTED_CONTENT_TYPES = {
+    "ReleaseCalendarPage": "release",
+    "StatisticalArticlePage": "bulletin",
+    "InformationPage": "static_page",
+    "IndexPage": "static_landing_page",
+    "MethodologyPage": "static_methodology",
+}
+
+
+class DummyPublisher(BasePublisher):
+    """Concrete subclass of BasePublisher for testing the shared functionality."""
+
+    def __init__(self, channel_created_or_updated, channel_deleted):
+        self._channel_created_or_updated = channel_created_or_updated
+        self._channel_deleted = channel_deleted
+
+    def _publish(self, channel, message):
+        # We don't actually publish in tests; we just want to spy on the calls.
+        pass
+
+    def get_created_or_updated_channel(self):
+        return self._channel_created_or_updated
+
+    def get_deleted_channel(self):
+        return self._channel_deleted
+
+
+class BasePublisherTests(TestCase, WagtailTestUtils):
+    @classmethod
+    def setUpTestData(cls):
+        cls.superuser = cls.create_superuser(username="admin")
+        # Pages that are NOT in SEARCH_INDEX_EXCLUDED_PAGE_TYPES
+        cls.included_pages = [
+            InformationPageFactory(),
+            MethodologyPageFactory(),
+            ReleaseCalendarPageFactory(),
+            StatisticalArticlePageFactory(),
+            IndexPageFactory(slug="custom-slug-1"),
+        ]
+
+        cls.release_calendar_page_provisional = ReleaseCalendarPageFactory(
+            status=ReleaseStatus.PROVISIONAL,
+            release_date=timezone.now() - timedelta(minutes=1),
+        )
+        cls.release_calendar_page_confirmed = ReleaseCalendarPageFactory(
+            status=ReleaseStatus.CONFIRMED,
+            release_date=timezone.now() - timedelta(minutes=1),
+        )
+        cls.release_calendar_page_cancelled = ReleaseCalendarPageFactory(
+            status=ReleaseStatus.CANCELLED,
+            release_date=timezone.now() - timedelta(minutes=1),
+        )
+        cls.release_calendar_page_published = ReleaseCalendarPageFactory(
+            status=ReleaseStatus.PUBLISHED,
+            release_date=timezone.now() - timedelta(minutes=1),
+        )
+
+    def setUp(self):
+        self.publisher = DummyPublisher(
+            channel_created_or_updated="dummy-channel-created",
+            channel_deleted="dummy-channel-deleted",
+        )
+
+        self.client.force_login(self.superuser)
+
+        self.root_page = Page.objects.get(id=1)
+        # self.root_topic = Topic.objects.root_topic()
```

Comment:
```suggestion
```

Reply:
🦅 👀 !

Reply:
Great spot! I was going to remove that line + other needless lines of code like that in taxonomy too when I wrote the tests there, and ended up leaving it in last time.

Removed now!